yang/goldstone-routing.yang

```diff
@@ -0,0 +1,95 @@
+module goldstone-routing {
+
+  yang-version "1";
+
+  namespace "http://goldstone.net/yang/goldstone-routing";
+  prefix gs-routing;
+
+  organization
+    "GoldStone";
+
+  contact
+    "Goldstone";
+
+  description
+    "Goldstone Routing";
+
+  revision 2022-12-14 {
+    description
+      "Initial revision.";
+  }
+
+  typedef control-plane-protocol {
+    type enumeration {
+      enum STATIC;
+    }
+  }
+
+  typedef administrative-distance {
+    type uint8 {
+      range "1..255";
+    }
+    description
+      "Admin distance associated with the route.";
+  }
+
+  grouping routing-common-config {
+    description
+      "Configuration data nodes common to the routing
+      subsystems";
+
+    leaf type {
+      type control-plane-protocol;
+      description
+        "Type of the control-plane protocol";
+    }
+
+    leaf name {
+      type string;
+      description
+        "An arbitrary name of the control-plane protocol
+         instance.";
+    }
+
+  }
+
+  container routing {
+    description
+      "Configuration parameters for the routing subsystem.";
+
+    container control-plane-protocols {
+      description
+        "Support for control-plane protocol instances.";
+
+      list control-plane-protocol {
+        key "type name";
```

Comment:
This enables having multiple control planes of the same type.
Could you explain the use case of this?
Does having two static routing protocol instances make sense?

Reply:
That is a valid concept. I am not sure how often it can be used but, for example, more than one OSPF instance can be configured and work independently from each other.

Reply:
@ishidawataru @ipi-claytonpascoal 
Thank you for pointing it out.
Sorry, I got too caught up on the extensibility for other protocols and FRR yang structure.
At least for the current stage, having two static routing protocol instances doesn't make sense. I will fix this part.

Reply:
> At least for the current stage, having two static routing protocol instances doesn't make sense. I will fix this part.

I agree. Let's consider supporting multiple control planes of the same type when we get an actual use case.